docs/project-development.md

```diff
@@ -260,19 +260,16 @@ Haskell process. This is recommended to allow testing different
 browsers, and to make use of a browser's significantly better
 developer tools.
 
-To use it, add `jsaddle-warp` and `reflex-dom-core` to your frontend's
-dependencies, and change `main` like so:
+To use it, enable the `use-warp` flag in `reflex-dom` by adding the 
+following to `default.nix` (right below its shells section):
 
-```haskell
-import Language.Javascript.JSaddle.Warp
-import Reflex.Dom.Core (mainWidget)
-import Reflex.Dom hiding (mainWidget, run)
-
-main :: IO ()
-main = run 3911 $ mainWidget app
+```nix
+  overrides = self: super: { 
+    reflex-dom = pkgs.haskell.lib.addBuildDepend (pkgs.haskell.lib.enableCabalFlag super.reflex-dom "use-warp") self.jsaddle-warp;
```

Comment:
Maybe:

```
reflex-dom =
  if self.ghc.isGhcjs || self.ghc.isCross
    then super.reflex-dom
    else pkgs.haskell.lib.addBuildDepend (pkgs.haskell.lib.enableCabalFlag super.reflex-dom "use-warp") self.jsaddle-warp;
```

Reply:
That looks like it might work. I see `isGhcjs` in reflex-platform; but there is no `isCross`.

Reply:
Oh wait, spoke too soon. That works! With a slight change: `if (self.ghc.isGhcjs or false) || (self.ghc.cross or null) == null`

Reply:
Grr.. nope. Something wrong here.

Reply:
Hm. Yea, I don't remember the right way to tell if a `ghc` is cross or not.

Reply:
In reflex-platform, I see:

```
            addReflexOptimizerFlag = if useReflexOptimizer && (self.ghc.cross or null) == null
              then drv: appendConfigureFlag drv "-fuse-reflex-optimizer"
              else drv: drv;
```

But the `cross` attribute is unavailable from default.nix.